### PR TITLE
YahooFinance downloader

### PR DESCRIPTION
**Pull request by CremaLuca**
## Premise

I wanted to integrate yahoo finance in the history downloader so I made an adapter for yfinance library.

## Changes

This class is able to perform the following actions:

- Download data for a given a period of time (chosen between allowed intervals) and an interval for data aggregation (see allowed intervals).
- Download data between two dates and an interval for data aggregation.
- Download data between two dates with the minimum interval possible.
- Download last week's data with 1m interval.

## Known flaws

bulk_download.py needs a lot of refactoring, but I won't do it in this PR.

### CODE REVIEW

File: av_downloader.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def download(self, symbol, interval='1min'):
                 return ""
 
 if __name__ == '__main__':
-    dir_path = Path(Path.cwd(), 'av_historical_1m')
+    dir_path = Path('/av_historical_1m')
```
Comment (ricdezen) [inline av_downloader.py:51]: Aight glad to see you figured it out, worked on my machine not sure why tho, maybe the directory was already there from a previous run and I did not notice

File: docs/russell3000.json [hunk 2]
```diff
@@ -1,12011 +1,12011 @@
 {
     "tickers":[
         {
-            "Ticker":"A",
+            "ticker":"A",
```
Comment (ricdezen) [inline docs/russell3000.json:4]: Fai enough

File: yf_downloader.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+
+class YFinanceDownloader:
+    '''
+    Attributes:
+        files_directory : pathlib.Path
+            Directory where to put JSON output files in.
+    
+    '''
+    def __init__(self, files_directory : Path):
```
Comment (ricdezen) [inline yf_downloader.py:18]: Ok I somehow missed from the Python book that arguments can be typed, good news

File: bulk_download.py [hunk 4]
```diff
@@ -61,19 +62,40 @@ def log_result(log_path, msg):
 
 if __name__ == '__main__':
 
-    print(Path.cwd())
+    api_choice = input("Aplha Vantage or Yahoo finance? A/Y")
 
-    av_key = input('Insert your Alpha Vantage api key: ')
+    if(api_choice == "A"):
```
Comment (ricdezen) [inline bulk_download.py:67]: Well yes we would need to refactor bulk_download.py, it was just meant to be a very specific main script

File: yf_downloader.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+    Parameters:
+        ticker : str
+            Output ticker's name.
+        period : str
+            Yahoo's period of time
+    '''
+    return "{}_{}_{}.json".format(ticker, date.today().strftime("%d-%m-%y"), period)
+
+def __calc_minimum_interval(start_date : date):
```
Comment (ricdezen) [inline yf_downloader.py:150]: I'm gonna make a wild guess and assume the double underscores mean "private". TIL
Edit: yep

**Review by ricdezen — APPROVED** [review]
Alright I've had a look around, everything seems in order, no particular issues, we'll just have to modularize bulk_downloader

File: yf_downloader.py (hunk 5 above)
Comment (CremaLuca) [reply yf_downloader.py:150]: These are private functions, they used to be private methods but made no use of the `self` parameter, so codacy suggested to move them to functions. I have no idea if there are such things as private and public functions of a module but I'll find out.

**Pull request merged**